flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClient.java

```diff
@@ -211,18 +185,45 @@ public void run() {
 		Runtime.getRuntime().addShutdownHook(clientShutdownHook);
 
 		isConnected = true;
+
+		logAndSysout("Waiting until all TaskManagers have connected");
```

Comment:
Since the `logAndSysout` method is defined in the `ClusterClient`, all log messages appear from being from that class, even though they are coming from the `YarnClusterClient`. Maybe it makes sense to override the method or add an argument for the logger.